workflow/rocoto/workflow_tasks.py

```diff
@@ -1102,57 +1086,52 @@ def eupd(self):
 
         return task
 
-    def atmensanalprep(self):
-
-        dump_suffix = self._base["DUMP_SUFFIX"]
+    def atmensanlinit(self):
         gfs_cyc = self._base["gfs_cyc"]
-        dmpdir = self._base["DMPDIR"]
         do_gfs_enkf = True if self.app_config.do_hybvar and 'gfs' in self.app_config.eupd_cdumps else False
+        cycledef = 'gdas'
+        if self.cdump in ['gfs'] and do_gfs_enkf:
+            cycledef = 'gfs'
+            if gfs_cyc != 4:
+                cycledef = 'gdas'
+        task_cdump = cycledef
 
         deps = []
-        dep_dict = {'type': 'metatask', 'name': 'gdaspost', 'offset': '-06:00:00'}
-        deps.append(rocoto.add_dependency(dep_dict))
-        data = f'&ROTDIR;/gdas.@Y@m@d/@H/atmos/gdas.t@Hz.atmf009.nc'
-        dep_dict = {'type': 'data', 'data': data, 'offset': '-06:00:00'}
+        dep_dict = {'type': 'task', 'name': f'{task_cdump}prep'}
         deps.append(rocoto.add_dependency(dep_dict))
-        data = f'{dmpdir}/{self.cdump}{dump_suffix}.@Y@m@d/@H/atmos/{self.cdump}.t@Hz.updated.status.tm00.bufr_d'
-        dep_dict = {'type': 'data', 'data': data}
+        dep_dict = {'type': 'metatask', 'name': 'enkfgdasepmn', 'offset': '-06:00:00'}
         deps.append(rocoto.add_dependency(dep_dict))
         dependencies = rocoto.create_dependency(dep_condition='and', dep=deps)
 
-        cycledef = self.cdump
-        if self.cdump in ['gfs'] and do_gfs_enkf and gfs_cyc != 4:
-            cycledef = 'gdas'
-
-        resources = self.get_resource('atmensanalprep')
-        task = create_wf_task('atmensanalprep', resources, cdump=self.cdump, envar=self.envars, dependency=dependencies,
+        resources = self.get_resource('atmensanlinit')
+        task = create_wf_task('atmensanlinit', resources, cdump=self.cdump, envar=self.envars, dependency=dependencies,
                               cycledef=cycledef)
 
         return task
 
-    def atmensanalrun(self):
+    def atmensanlrun(self):
```

Comment:
Do these other tasks in the atmensanl job sequence need to run in the same `cycledefs` as the `init`?

Reply:
Yes, `atmensanlinit`, `atmensanlrun`, and `atmensanlfinal` are an initialize, run, finalize group which need to sequentially run in the same cycle.    Not sure this answers your question.